protocol_rfcs/managed-commits.md

```diff
@@ -0,0 +1,315 @@
+# Managed Commits
+**Associated Github issue for discussions: https://github.com/delta-io/delta/issues/2598**
+
+This RFC proposes a new table feature `managedCommits` which changes the way Delta Lake manages commits.
+
+Today’s Delta commit protocol relies on the filesystem to provide commit atomicity. This feature request is to allow delta tables which gets commit atomicity using an external commit-owner and not
+the filesystem (s3, abfs etc). This allows us to deal with various limitations of delta:
+
+1. No reliable way for the table's owner to participate in commits.
+    - The table's owner (such as a catalog) cannot reliably stay in sync with the table state, nor reject commit attempts it wouldn’t like, because it doesn’t even know about writes until they are already durable (and visible to readers).
+    - No clear path to transactions that could span multiple tables and/or involve catalog updates, because filesystem commits cannot be made conditionally or atomically.
+2. No way to tie commit ownership to a table.
+    - In general, Delta tables have no way to advertise that they are managed by catalog or LogStore X (at endpoint Y)
+    - There is no central entity that needs to be contacted in order to commit to the table. So if the underlying file system is missing _putIfAbsent_ semantics, then there is no way to ensure that a commit is atomic, which could lead
+      to lost writes when concurrent writers are writing to the table.
+
+--------
+
+> ***Change to existing section***
+
+### Delta Log Entries
+Delta files are stored as JSON in a directory at the root of the table named `_delta_log`, and together with checkpoints make up the log of all changes that have occurred to a table. They are the unit of atomicity for a table.
+
+**Note:** If [managed commits](#managed-commits) table feature is enabled on the table, recently committed delta files may reside in the `_delta_log/_commits` directory. Delta clients have to contact
+the corresponding commit-owner of the table in order to find the information about the [un-backfilled commits](#commit-backfills).
```

Comment:
Yes it was creating confusion as we are trying to simulate a diff. So now this just reflects final state of this section. i.e. how it looks now.

Reply:
That makes it hard to review for _changes_ tho... for example, the old text said something about commit files being the "unit of atomicity" for the table, and that's completely gone in this version. But nothing actually identifies that change. 

I did notice that mixing lots of <ins>insertions</ins> and ~deletions~ in the same text is hard to read, and it will often be better to just put full before/after versions, e.g. 

~Delta files are the unit of atomicity for a table, and are named using the next available version number, zero-padded to 20 digits.~

<ins>The delta files in `_delta_log` directory are named using the next available version number, zero-padded to 20 digits.</ins>

But any time we're reviewing markdown changes, it's important to View File, to see how it actually renders instead of trying to guess from the raw text:
![image](https://github.com/delta-io/delta/assets/79601771/64a38241-d629-4047-af9a-d4e859657a78)

Reply:
fixed.